server/field/lifepool/mob/mob.go

```diff
@@ -471,6 +477,25 @@ func (m Data) String() string {
 
 // Update mob for status changes e.g. posion, hp/mp recover, finding a new controller after inactivity
 func (m *Data) Update(t time.Time) {
+	checkTime := t.Unix()
+	m.lastStatusUpdate = checkTime
+
+	if m.hp <= 0 {
+		return
+	}
+
+	if m.poison {
+		// Handle poison
+		m.hp = m.hp - 10 // Need to adjust to poison amount based on poison level
+	}
+
+	// Update mob status if one is applied
+	if (checkTime - m.lastHeal) > 30 {
+		// Heal the mob
+		regenhp, regenmp := int32(10), int32(10) // Need to figure out how to find these values? Are they set in nx file for mob?
```

Comment:
I've looked around but I can't find what these values should be? I can make an algorithm if nothing else. Maybe heal 0.01% or so every 30 seconds? I am not too sure what these numbers should look like.

Reply:
No problem, might have to look at other sources for that. I didn't know mobs could heal hp passively!

When you are ready for this branch to be merged in leave a comment.

Reply:
Alright sounds good. I will continue to dig on it, and if I do not find anything I will just put together a simple algorithm for now. 

I believe the HP heal might be a bit more difficult to implement if I cannot find values. Obviously we don't want the mob healing if it is actively being attacked so I will need to add checks for that as well. MP should be rather simple because it should not matter whether or not it is being attacked.

Reply:
I think for hp it should be based on time since it was last attacked? Maybe have something like: After 60 seconds of not being attacked hp is restored to full? I can't remember if the original game had hp regen?

Reply:
I think full hp after 60 seconds might be too aggressive. It would be not great especially for bosses or high hp mobs. I've set it to restore 1% after 60 seconds every 30 seconds. I believe this should be a fair heal amount for the time being.